pkg/api/correlation.go

```diff
@@ -1,12 +1,22 @@
 package api
 
 import (
+	"context"
+	"net/http"
 	"time"
+
+	"github.com/Azure/ARO-RP/pkg/util/uuid"
 )
 
 // Copyright (c) Microsoft Corporation.
 // Licensed under the Apache License 2.0.
 
+type contextKey int
+
+const (
+	contextKeyCorrelationData contextKey = iota
```

Comment:
What is the purpose of using `iota` here?

Reply:
We could use an integer, but using iota we have the benefit that if we add more const types below _contextKeyCorrelationData_ it will automatically have an int assigned. It is just about making it more extensible. Without the iota we need to manually assign a number for this constant key and the new ones in the future. It could be the literal 0, though. I don't see any drawback about using the iota here (I could be wrong). 🙂 

The important thing is to provide a function at package level that allows you to create a context with the given value (without exposing the key used as it is just an implementation detail that could change in the future).